src/Symfony/Component/Messenger/Transport/Schedule/Schedule.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport\Schedule;
+
+use Symfony\Component\Messenger\Transport\Schedule\Progress\ProgressInterface;
+use Symfony\Component\Messenger\Transport\Schedule\Trigger\TriggerInterface;
+
+final class Schedule implements ScheduleInterface
+{
+    /**
+     * @var array<int, array{TriggerInterface, object}>
+     */
+    private array $schedule;
+    private ProgressInterface $progress;
+    private ScheduleHeap $scheduleHeap;
+
+    public function __construct(ScheduleConfig $scheduleConfig, ProgressInterface $progress)
```

Comment:
```suggestion
    public function __construct(
        ScheduleConfig $scheduleConfig,
        private ProgressInterface $progress,
    )
```

Reply:
Ok... But IMO, mixing promoted properties and regular ones is ugly.